processor/metricstransformprocessor/operation_aggregate_label_values.go

```diff
@@ -54,7 +54,9 @@ func (mtp *metricsTransformProcessor) groupTimeseriesByNewLabelValue(metric *met
 		}
 
 		key, newLabelValues := mtp.newLabelValuesAsKey(labelIdx, timeseries, newValue, aggregatedValuesSet)
-		key += strconv.FormatInt(timeseries.StartTimestamp.Seconds, 10)
+		if timeseries.StartTimestamp != nil {
+			key += strconv.FormatInt(timeseries.StartTimestamp.Seconds, 10)
```

Comment:
Do you know if the uniqueness of `key` would be compromised if we don't append the start time? Would label values alone be unique enough, maybe along with the metric name?

Reply:
I believe this is fine. The points within the timeseries will only be aggregated if they have the same end timestamp. Also note this edge case will go away when we switch to the OTLP proto which doesn't have the concept of timeseries iiuc.